### PR TITLE
Bug 1905502: Polling route until registry is available

### PR DESCRIPTION
**Pull request by ricardomaraschini**
This PR adds a check on the external route for the ephemeral registry right after its deployment, this way we can assure the registry is accessible from outside the cluster prior to tests execution.

Networking team uses the following script to achieve the same on their tests:

https://github.com/openshift/origin/blob/master/test/extended/router/scoped.go#L241-L271

### CODE REVIEW

File: pkg/testframework/registry.go [hunk 1]
```diff
@@ -482,3 +492,39 @@ func CreateEphemeralRegistry(t *testing.T, restConfig *rest.Config, namespace st
...
+
+	url := fmt.Sprintf("https://%s/v2/", host)
+	client := &http.Client{Transport: tr}
+	res, err := client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode == http.StatusServiceUnavailable {
```
Comment (adambkaplan) [inline pkg/testframework/registry.go:522]: Do we care if the status code is something else, like 404?

In reviewing what the networking team does, their script only exits 0 if the OK (200) response is returned.

Comment (ricardomaraschini) [reply pkg/testframework/registry.go]: Yes, you are right. Adjusted accordingly.

Comment (ricardomaraschini) [reply pkg/testframework/registry.go]: Just clarifying this, for some tests the registry is deployed with basic auth in place. In such scenario a 403 return is also OK. I have adjusted the deployment to consider 200 or 403 as "the registry is good to go".

File: pkg/testframework/registry.go [hunk 2]
```diff
@@ -482,3 +492,39 @@ func CreateEphemeralRegistry(t *testing.T, restConfig *rest.Config, namespace st
...
+
+	url := fmt.Sprintf("https://%s/v2/", host)
+	client := &http.Client{Transport: tr}
+	res, err := client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusUnauthorized {
```
Comment (adambkaplan) [inline pkg/testframework/registry.go:522]: nit - add a comment that `http.StatusUnauthorized` indicates that the registry is operating normally.

**Review by adambkaplan — CHANGES_REQUESTED** [review]

File: pkg/testframework/registry.go (hunk 2 above)
Comment (ricardomaraschini) [reply pkg/testframework/registry.go:524]: Good call. Done.

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**